airflow/upgrade/rules/postgres_mysql_sqlite_version_upgrade_check.py

```diff
@@ -0,0 +1,61 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+from airflow.configuration import conf
+from airflow.upgrade.rules.base_rule import BaseRule
+from airflow.utils.db import provide_session
+
+
+class DatabaseVersionCheckRule(BaseRule):
+
+    title = "Check versions of PostgreSQL, MySQL, and SQLite to ease upgrade to Airflow 2.0"
+
+    description = """\
+From Airflow 2.0, the following database versions are supported:
+PostgreSQl - 9.6, 10, 11, 12, 13;
+MySQL - 5.7, 8;
+SQLite - 3.15+
+    """
+
+    @provide_session
+    def check(self, session=None):
+
+        more_info = "See link below for more details: https://github.com/apache/airflow#requirements"
+
+        conn_str = conf.get(section="core", key="sql_alchemy_conn")
+
+        if "sqlite" in conn_str:
+            result = session.execute('select sqlite_version();').fetchone()[0]
+            if int(result.split('.')[0]) < 3:
+                return "SQLite version below 3.15 not supported. \n" + more_info
+            elif int(result.split('.')[0]) == 3 and int(result.split('.')[1]) < 15:
+                return "SQLite version below 3.15 not supported. \n" + more_info
```

Comment:
You can compare the version in a better way:

```python
 from packaging.version import Version
try:
    import importlib.metadata as importlib_metadata
except ImportError:
    import importlib_metadata

min_req_sqlite_vesion = Version('3.15')
installed_sqlite_version =  Version(session.execute('select sqlite_version();').scalar())
if installed_sqlite_version < min_req_sqlite_vesion:
    return "SQLite version below 3.15 not supported. \n" + more_info
```

Reply:
Example:

```python
In [5]: session.execute('select sqlite_version();').scalar()
Out[5]: '3.30.0'

In [6]: import importlib_metadata

In [7]: from packaging.version import Version

In [8]: Version('3.30.0')
Out[8]: <Version('3.30.0')>

In [9]: Version('3.30.0') < Version('3.15')
Out[9]: False

In [10]: Version('3.3.0') < Version('3.15')
Out[10]: True
```

Reply:
oh cool!